GeoLib/Grid.h

```diff
@@ -201,6 +201,10 @@ Grid<POINT>::Grid(InputIterator first, InputIterator last,
                                     _max_pnt[1] - _min_pnt[1],
                                     _max_pnt[2] - _min_pnt[2]}};
 
+    // enlarge delta
+    for (auto & d : delta)
+        d = std::nextafter(d, std::numeric_limits<double>::max());
```

Comment:
Yes, `delta` is the space diagonal of the bounding box and also of the grid. The bounding box itself is extended already such a tiny bit during its construction. But when `delta` is computed from the bounding box the tiny extension of the bounding box could get lost because of numerical precision. So here the `delta` is enlarged again such that all points fits into the grid.